controllers/userControllers.js

```diff
@@ -1,101 +1,94 @@
-const User = require("../models/User");
-//const bcryptjs = require("bcryptjs");
-
-/**
- * things to think about
- * how we grab users : through req.body or req.params
- *
- */
+const User = require('../models/User');
+const mongoose = require('mongoose');
+const customError = require('../helpers/customErrorHandler');
 
 exports.getUsers = async (req, res) => {
   let users = await User.find();
   res.json(users);
 };
 
 exports.getUser = async (req, res, next) => {
-  // Grab the id from the url
   const { id } = req.params;
-  // Find the user with that id
   try {
     let user = await User.findById(id);
+    // id is a valid mongoose id
+    if (!user) {
+      next(customError(`User with ID: ${id} does not exist`, 400));
+    }
     res.json(user);
   } catch (err) {
-    //todo : are we making custom errors?
-    next(err); // found error of not found todo
+    // id is not a valid mongoose id
+    if (err instanceof mongoose.Error.CastError) {
+      next(customError(`ID: ${id} is not valid`, 400));
+    }
+    next(err);
   }
 };
 
-exports.updateUser = async (req, res) => {
-  // Grab the id from the url
+exports.updateUser = async (req, res, next) => {
+  // todo : check the req.body to see if the information is good and there is a change
   const { id } = req.params;
-  // todo : do we have a middle ware to check the req.body to see if the information is good or allowed?
+
   try {
-    let user = await User.findById(id); // update the user fields
+    let user = await User.findById(id);
+    if (!user) {
+      next(customError(`User with ID: ${id} does not exist`, 400));
+      return;
```

Comment:
you need to return here as well?

Reply:
yes, because we get a result back, if there is no return the code will run till the end and take the res.json